pyomo/contrib/doe/fim_doe.py

```diff
@@ -0,0 +1,2582 @@
+        ax.set_xticks(range(len(xLabel)))
+        ax.set_xticklabels(xLabel)
+        ax.set_xlabel(xlabel_text)
+        im = ax.imshow(hes_a.T, cmap=plt.pyplot.cm.hot_r)
+        ba = plt.pyplot.colorbar(im)
+        ba.set_label('log10(trace(FIM))')
+        plt.pyplot.title(title_text + ' - A optimality')
+        plt.pyplot.show()
+
+        # D-optimality
+        fig = plt.pyplot.figure()
+        plt.pyplot.rc('axes', titlesize=font_axes)
+        plt.pyplot.rc('axes', labelsize=font_axes)
+        plt.pyplot.rc('xtick', labelsize=font_tick)
+        plt.pyplot.rc('ytick', labelsize=font_tick)
+        ax = fig.add_subplot(111)
+        params = {'mathtext.default': 'regular'}
+        plt.pyplot.rcParams.update(params)
+        ax.set_yticks(range(len(yLabel)))
+        ax.set_yticklabels(yLabel)
+        ax.set_ylabel(ylabel_text)
+        ax.set_xticks(range(len(xLabel)))
+        ax.set_xticklabels(xLabel)
+        ax.set_xlabel(xlabel_text)
+        im = ax.imshow(hes_d.T, cmap=plt.pyplot.cm.hot_r)
+        ba = plt.pyplot.colorbar(im)
+        ba.set_label('log10(det(FIM))')
+        plt.pyplot.title(title_text + ' - D optimality')
+        plt.pyplot.show()
+
+        # E-optimality
+        fig = plt.pyplot.figure()
+        plt.pyplot.rc('axes', titlesize=font_axes)
+        plt.pyplot.rc('axes', labelsize=font_axes)
+        plt.pyplot.rc('xtick', labelsize=font_tick)
+        plt.pyplot.rc('ytick', labelsize=font_tick)
+        ax = fig.add_subplot(111)
+        params = {'mathtext.default': 'regular'}
+        plt.pyplot.rcParams.update(params)
+        ax.set_yticks(range(len(yLabel)))
+        ax.set_yticklabels(yLabel)
+        ax.set_ylabel(ylabel_text)
+        ax.set_xticks(range(len(xLabel)))
+        ax.set_xticklabels(xLabel)
+        ax.set_xlabel(xlabel_text)
+        im = ax.imshow(hes_e.T, cmap=plt.pyplot.cm.hot_r)
+        ba = plt.pyplot.colorbar(im)
+        ba.set_label('log10(minimal eig(FIM))')
+        plt.pyplot.title(title_text + ' - E optimality')
+        plt.pyplot.show()
+
+        # modified E-optimality
+        fig = plt.pyplot.figure()
+        plt.pyplot.rc('axes', titlesize=font_axes)
+        plt.pyplot.rc('axes', labelsize=font_axes)
+        plt.pyplot.rc('xtick', labelsize=font_tick)
+        plt.pyplot.rc('ytick', labelsize=font_tick)
+        ax = fig.add_subplot(111)
+        params = {'mathtext.default': 'regular'}
+        plt.pyplot.rcParams.update(params)
+        ax.set_yticks(range(len(yLabel)))
+        ax.set_yticklabels(yLabel)
+        ax.set_ylabel(ylabel_text)
+        ax.set_xticks(range(len(xLabel)))
+        ax.set_xticklabels(xLabel)
+        ax.set_xlabel(xlabel_text)
+        im = ax.imshow(hes_e2.T, cmap=plt.pyplot.cm.hot_r)
+        ba = plt.pyplot.colorbar(im)
+        ba.set_label('log10(cond(FIM))')
+        plt.pyplot.title(title_text + ' - Modified E-optimality')
+        plt.pyplot.show()
+
+
+
+
+def simulate_discretize_model(m,NFE,collo=True,initialize=True):
```

Comment:
Should this be moved to the examples? The choice of integration scheme is problem specific.